### PR TITLE
Refactor use of pytest-aiohttp plugins

### PR DESCRIPTION
**Pull request by sebastian-aranda**
This PR makes a refactor on the way of using the `pytest-aiohttp` plugin fixtures as detailed in the [documentation](https://docs.aiohttp.org/en/stable/testing.html#pytest). Also:

- `on_cleanup` methods were adjusted to avoid teardown errors.
- `asyncio_mode` was set to `strict` to allow using `pytest-aiohttp` plugins.
- `pytest_plugins = 'aiohttp.pytest_plugin'` was set to avoid haiving to install `pytest-aiohttp` as a dependency.
- `test_utils` was added to share resources across tests files.

### CODE REVIEW

File: python/love/commander/commands.py [hunk 1]
```diff
@@ -92,7 +93,14 @@ async def start_cmd(request):
 
     cmd.router.add_post("/", start_cmd)
 
-    async def on_cleanup(cmd):
+    async def on_cleanup(cmd_app):
+        """Close the remotes when cleaning the application
```
Comment (wvreeven) [inline python/love/commander/commands.py:97]: Please terminate the sentence in a `.`.

File: python/love/commander/commands.py [hunk 2]
```diff
@@ -92,7 +93,14 @@ async def start_cmd(request):
...
+
+        Parameters
+        ----------
+        cmd_app : object
```
Comment (wvreeven) [inline python/love/commander/commands.py:101]: Please enclose the argument type in back quotes.

File: python/love/commander/heartbeats.py [hunk 3]
```diff
@@ -45,7 +46,14 @@ async def start_heartbeat(request):
 
     heartbeat.router.add_get("/", start_heartbeat)
 
-    async def on_cleanup(heartbeat):
+    async def on_cleanup(heartbeat_app):
+        """Close the remotes when cleaning the application
```
Comment (wvreeven) [inline python/love/commander/heartbeats.py:50]: Please terminate the sentence in a `.`.

File: python/love/commander/heartbeats.py [hunk 4]
```diff
@@ -45,7 +46,14 @@ async def start_heartbeat(request):
...
+
+        Parameters
+        ----------
+        heartbeat_app : object
```
Comment (wvreeven) [inline python/love/commander/heartbeats.py:54]: Please enclose the argument type in back quotes.

File: python/love/commander/lovecsc.py [hunk 5]
```diff
@@ -105,6 +106,13 @@ async def post_observing_log(request):
     lovecsc_app.router.add_post("/observinglog", post_observing_log)
 
     async def on_cleanup(lovecsc_app):
+        """Close the CSC when cleaning the application
+
+        Parameters
+        ----------
+        lovecsc_app : object
```
Comment (wvreeven) [inline python/love/commander/lovecsc.py:113]: Please enclose the argument type in back quotes.

File: python/love/commander/lovecsc.py [hunk 6]
```diff
@@ -31,7 +32,7 @@
 
 
 def create_app(*args, **kwargs):
-    """Create the LOVECsc application
+    """Create the LOVE CSC application
```
Comment (wvreeven) [inline python/love/commander/lovecsc.py:35]: Please terminate the sentence in a `.`.

File: python/love/commander/lovecsc.py [hunk 7]
```diff
@@ -105,6 +106,13 @@ async def post_observing_log(request):
     lovecsc_app.router.add_post("/observinglog", post_observing_log)
 
     async def on_cleanup(lovecsc_app):
+        """Close the CSC when cleaning the application
```
Comment (wvreeven) [inline python/love/commander/lovecsc.py:109]: Please terminate the sentence in a `.`.

File: python/love/commander/reports.py [hunk 8]
```diff
@@ -160,7 +160,8 @@ def act(index, actuator) -> int:
     reports_app.router.add_post("/m1m3-bump-tests", query_m1m3_bump_tests)
     reports_app.router.add_post("/m1m3-bump-tests/", query_m1m3_bump_tests)
 
-    async def on_cleanup():
+    async def on_cleanup(reports_app):
+        # Do cleanup
```
Comment (wvreeven) [inline python/love/commander/reports.py:164]: Very good practice to add a comment here but please let the comment explain why the method is empty. Something like `This code doesn't require cleaning up.` or so would do.

File: python/love/commander/salinfo.py [hunk 9]
```diff
@@ -42,6 +41,7 @@ def create_app(*args, **kwargs):
         available_component_names = available_component_names[
             : kwargs.get("remotes_len_limit")
         ]
+
     salinfos = {}
```
Comment (wvreeven) [inline python/love/commander/salinfo.py:45]: In English the word `info` seldomly is used in plural form. A better name would be `salinfo_list`. Feel free to ignore this comment completely!

**Review by wvreeven — APPROVED** [review]
LGTM. I left a few comments for your consideration.

File: CHANGELOG.rst [hunk 10]
```diff
@@ -5,6 +5,7 @@ Version History
 v6.1.2
 ------
 
+* Refactor way of using pytest-aiohttp plugins `<https://github.com/lsst-ts/LOVE-commander/pull/71>`_
```
[the remark this answers is not in the document]
Comment (sebastian-aranda) [reply CHANGELOG.rst:8]: I like your suggestion! I'll apply it :+1:.

File: python/love/commander/salinfo.py (hunk 9 above)
Comment (sebastian-aranda) [reply python/love/commander/salinfo.py:45]: Nice! I'll apply this suggestion too :+1:.

**Pull request merged**